test/tests/functional/pbs_cray_reliable_job_startup.py

```diff
@@ -0,0 +1,200 @@
+    v = e.vnode_list[vn]
+    pbs.logjobmsg(e.job.id, "prologue: found vnode_list[" + v.name + "]")
+# print out the vnode_list_fail[] values
+for vn in e.vnode_list_fail:
+    v = e.vnode_list_fail[vn]
+    pbs.logjobmsg(e.job.id, "prologue: found vnode_list_fail[" + v.name + "]")
+if e.job.in_ms_mom():
+    pj = e.job.release_nodes(keep_select=e.job.Resource_List["site"])
+    if pj is None:
+        e.job.Hold_Types = pbs.hold_types("s")
+        e.job.rerun()
+        e.reject("unsuccessful at PROLOGUE")
+"""
+
+        # launch hook
+        self.launch_hook_body = """
+import pbs
+e=pbs.event()
+if 'PBS_NODEFILE' not in e.env:
+    e.accept()
+pbs.logmsg(pbs.LOG_DEBUG, "Executing launch")
+# print out the vnode_list[] values
+for vn in e.vnode_list:
+    v = e.vnode_list[vn]
+    pbs.logjobmsg(e.job.id, "launch: found vnode_list[" + v.name + "]")
+# print out the vnode_list_fail[] values:
+for vn in e.vnode_list_fail:
+    v = e.vnode_list_fail[vn]
+    pbs.logjobmsg(e.job.id, "launch: found vnode_list_fail[" + v.name + "]")
+    v.state = pbs.ND_OFFLINE
+if e.job.in_ms_mom():
+    pj = e.job.release_nodes(keep_select=e.job.Resource_List["site"])
+    if pj is None:
+        e.job.Hold_Types = pbs.hold_types("s")
+        e.job.rerun()
+        e.reject("unsuccessful at LAUNCH")
+"""
+
+    def match_str_in_input_file(self, file_path, file_pattern, search_str):
+        """
+        Assert that search string appears in the input file
+        that matches file_pattern
+        """
+        input_file = None
+        for item in self.du.listdir(path=file_path, sudo=True):
+            if fnmatch.fnmatch(item, file_pattern):
+                input_file = item
+                break
+        self.assertTrue(input_file is not None)
+        with PBSLogUtils().open_log(input_file, sudo=True) as f:
+            self.assertTrue(search_str in f.read())
+            self.logger.info("Found \"%s\" in %s" % (search_str, input_file))
+
+    @tags('cray', 'smoke')
+    def test_reliable_job_startup_not_supported_on_cray(self):
+        """
+        A job is started with extra nodes. Mom superior will show no sign
+        of tolerating node failure.  Accounting logs won't have 's' record.
+        Input files to prologue and launch hooks will show the
+        tolerate_node_failures=none value.
+        """
+        # instantiate queuejob hook
+        hook_event = "queuejob"
+        hook_name = "qjob"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.qjob_hook_body)
+
+        # instantiate execjob_prologue hook
+        hook_event = "execjob_prologue"
+        hook_name = "prolo"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.prolo_hook_body)
+
+        # instantiate execjob_launch hook
+        hook_event = "execjob_launch"
+        hook_name = "launch"
+        a = {'event': hook_event, 'enabled': 'true'}
```

Comment:
Your use of quotes should be consistent throughout the file. Why do you use single quotes in some cases and double quotes in others?

Reply:
Done.

Reply:
Most instances of double quotes have been replaced with single quotes. I think it's sufficient.